app/models/manageiq/providers/vmware/infra_manager/inventory/parser.rb

```diff
@@ -232,14 +232,12 @@ def parse_network(object, kind, props)
   alias parse_opaque_network parse_network
 
   def parse_distributed_virtual_portgroup(_object, kind, props)
-    return if kind == "leave"
```

Comment:
This is the core of the fix, the `"leave"` action deletes the portgroup from the cache [[here](https://github.com/ManageIQ/manageiq-providers-vmware/blob/f33e6db3afbcd70d40b58088591ce63901a49ab2/app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb#L235-L237)], but since lans are subordinate to switches we can't just refresh a single lan so we loop through all the lans on the switch [[here](https://github.com/ManageIQ/manageiq-providers-vmware/blob/f33e6db3afbcd70d40b58088591ce63901a49ab2/app/models/manageiq/providers/vmware/infra_manager/inventory/parser.rb#L120-L125)].  With the deleted lan not in the cache this lan isn't built and the old one is deleted from the database.